src/voice/VoiceRecorder.ts

```diff
@@ -130,23 +130,18 @@ export class VoiceRecorder {
         // The time domain is the input to the FFT, which means we use an array of the same
         // size. The time domain is also known as the audio waveform. We're ignoring the
         // output of the FFT here (frequency data) because we're not interested in it.
-        //
-        // We use bytes out of the analyser because floats have weird precision problems
-        // and are slightly more difficult to work with. The bytes are easy to work with,
-        // which is why we pick them (they're also more precise, but we care less about that).
-        const data = new Uint8Array(this.recorderFFT.fftSize);
-        this.recorderFFT.getByteTimeDomainData(data);
-
-        // Because we're dealing with a uint array we need to do math a bit differently.
-        // If we just `Array.from()` the uint array, we end up with 1s and 0s, which aren't
-        // what we're after. Instead, we have to use a bit of manual looping to correctly end
-        // up with the right values
+        const data = new Float32Array(this.recorderFFT.fftSize);
+        this.recorderFFT.getFloatTimeDomainData(data);
+
+        // We can't just `Array.from()` the array because we're dealing with 32bit floats
+        // and the built-in function won't consider that when converting between numbers.
+        // However, the runtime will convert the float32 to a float64 during the math operations
+        // which is why the loop works below. Note that a `.map()` call also doesn't work
+        // and will instead return a Float32Array still.
         const translatedData: number[] = [];
         for (let i = 0; i < data.length; i++) {
-            // All we're doing here is inverting the amplitude and putting the metric somewhere
-            // between zero and one. Without the inversion, lower values are "louder", which is
-            // not super helpful.
-            translatedData.push(1 - (data[i] / 128.0));
+            // We're clamping the values so we can do that math operation mentioned above.
```

Comment:
According to the comment above this is a js hack to convert 32 bit floats to 64 bit floats, but then this comment says its clamping them (when it appears to just be dividing by 1?)

Reply:
ah, we're doing both. Will clarify.